src/policy/packages.cpp

```diff
@@ -92,3 +91,81 @@ bool IsChildWithParents(const Package& package)
     return std::all_of(package.cbegin(), package.cend() - 1,
                        [&input_txids](const auto& ptx) { return input_txids.count(ptx->GetHash()) > 0; });
 }
+
+// After calling visit() on a package tx, it's guaranteed to be in the ancestor_subsets map.
+void Packageifier::visit(const CTransactionRef& curr_tx)
+{
+    const uint256& curr_txid = curr_tx->GetHash();
+    if (ancestor_subsets.count(curr_txid) > 0) return;
+    std::set<uint256> my_ancestors;
+    my_ancestors.insert(curr_txid);
+    for (const auto& input : curr_tx->vin) {
+        auto parent_tx = txid_to_tx.find(input.prevout.hash);
+        if (parent_tx == txid_to_tx.end()) continue;
+        if (ancestor_subsets.count(parent_tx->first) == 0) {
```

Comment:
You are right. Not sure if it's worth commenting. It's unlikely someone will violate this assumption (it would require big changes to Bitcoin), but it could save time to someone like me who forget about this property.

Reply:
I've added a comment to visit()